### PR TITLE
[OBSDEF-13968] Adding support for icons in modal title

### PR DESCRIPTION
**Pull request by tejasjc-ecs**
Adding support for icons in modal title

stbicon

### CODE REVIEW

File: src/modals/Modal.tsx [hunk 1]
```diff
@@ -106,8 +112,44 @@ export class Modal extends React.PureComponent<ModalProps> {
...
+        }
+    }
+
+    /*Function to build title for modal
+      overwrites type icons if custom icon is provided, doesn't renders any icon if showIcon is set to false
+    */
+    buildTitle(): React.ReactElement {
+        const {type, customIcon, title} = this.props;
+        const {showIcon} = this.state;
+        let titleIcon: React.ReactNode | undefined = customIcon
```
Comment (akate1) [inline src/modals/Modal.tsx:134]: let -> const

File: src/modals/Modal.tsx [hunk 2]
```diff
@@ -106,8 +112,44 @@ export class Modal extends React.PureComponent<ModalProps> {
...
+    getTitleIconByType(type: ModalType): React.ReactNode {
+        switch (type) {
+            case ModalType.INFO:
+                return <Icon aria-hidden={true} shape="info-standard" />;
+            case ModalType.WARNING:
+                return <Icon aria-hidden={true} shape="warning-standard" />;
+            case ModalType.DANGER:
+                return <Icon aria-hidden={true} shape="error-standard" />;
+            default:
+                return;
```
Comment (akate1) [inline src/modals/Modal.tsx:124]:
```suggestion
                return undefined;
```

File: src/modals/Modal.tsx [hunk 3]
```diff
@@ -106,8 +112,44 @@ export class Modal extends React.PureComponent<ModalProps> {
...
+      overwrites type icons if custom icon is provided, doesn't renders any icon if showIcon is set to false
+    */
+    buildTitle(): React.ReactElement {
+        const {type, customIcon, title} = this.props;
+        const {showIcon} = this.state;
+        let titleIcon: React.ReactNode | undefined = customIcon
+            ? customIcon
+            : type && type !== ModalType.DEFAULT
+            ? this.getTitleIconByType(type)
+            : undefined;
```
Comment (akate1) [inline src/modals/Modal.tsx:138]:
```suggestion
        const titleIcon: React.ReactNode | undefined = customIcon
            ? customIcon :        
             this.getTitleIconByType(type)
           
```

File: src/modals/Modal.tsx [hunk 4]
```diff
@@ -38,10 +40,13 @@ type ModalProps = {
     height?: number;
     className?: string;
     type?: ModalType;
+    showIcon?: boolean;
+    customIcon?: React.ReactNode;
 };
 
 type ModalState = {
     isOpen: boolean;
+    showIcon?: boolean;
```
[the remark this answers is not in the document]
Comment (tejasjc-ecs) [reply src/modals/Modal.tsx]: Removed showIcon from state

File: src/modals/Modal.tsx [hunk 5]
```diff
@@ -106,8 +112,44 @@ export class Modal extends React.PureComponent<ModalProps> {
...
+                return <Icon aria-hidden={true} shape="error-standard" />;
+            default:
+                return;
+        }
+    }
+
+    /*Function to build title for modal
+      overwrites type icons if custom icon is provided, doesn't renders any icon if showIcon is set to false
+    */
+    buildTitle(): React.ReactElement {
```
[the remark this answers is not in the document]
Comment (tejasjc-ecs) [reply src/modals/Modal.tsx]: Done

File: src/modals/Modal.tsx [hunk 6]
```diff
@@ -106,8 +112,44 @@ export class Modal extends React.PureComponent<ModalProps> {
...
+        const {type, customIcon, title} = this.props;
```
[the remark this answers is not in the document]
Comment (tejasjc-ecs) [reply src/modals/Modal.tsx]: Done

File: src/modals/Modal.tsx [hunk 7]
```diff
@@ -106,8 +112,44 @@ export class Modal extends React.PureComponent<ModalProps> {
...
+        const {showIcon} = this.state;
```
[the remark this answers is not in the document]
Comment (tejasjc-ecs) [reply src/modals/Modal.tsx]: Done

File: src/modals/Modal.tsx (hunk 1 above)
Comment (tejasjc-ecs) [reply src/modals/Modal.tsx]: Done

File: src/modals/Modal.tsx (hunk 2 above)
Comment (tejasjc-ecs) [reply src/modals/Modal.tsx]: Done

File: src/modals/Modal.tsx (hunk 3 above)
Comment (tejasjc-ecs) [reply src/modals/Modal.tsx]: The ModalType defualt need not to have any Icons hence this condition is added, if modal type is set as default no pre defined icon is set

Comment (akate1) [reply src/modals/Modal.tsx]: @tejasjc-ecs  My suggested code block will also work with Default Type. can u give a try ?

Comment (tejasjc-ecs) [reply src/modals/Modal.tsx]: Yes, the suggested code worked as expected as default case returning undefined already, thanks! Done this change

**Review by akate1 — APPROVED** [review]

**Pull request merged**